cli/tests/unit/fetch_test.ts

```diff
@@ -54,9 +54,31 @@ function findClosedPortInRange(
   );
 }
 
+function flakyTest(
+  fn: (t: Deno.TestContext) => Promise<void>,
+) {
+  return async function (t: Deno.TestContext) {
+    const NO_OF_TRIES = 3;
+    let tries = 0;
+    let lastError;
+
+    while (tries < NO_OF_TRIES) {
+      try {
+        await fn(t);
+        return;
+      } catch (e) {
+        tries++;
+        lastError = e;
+      }
+    }
+
+    throw lastError;
```

Comment:
Nitpick: Maybe something like this?

```suggestion
    let lastError;

    for (let i = 0; i < 3; i++) {
      try {
        await fn(t);
        return;
      } catch (e) {
        lastError = e;
      }
    }
    
    throw lastError;
```